test/boost/service_level_controller_test.cc

```diff
@@ -61,6 +65,10 @@ struct qos_configuration_change_suscriber_simple : public qos_configuration_chan
         ops.push_back(change_op{sl_info.name, slo_before, slo_after});
         return make_ready_future<>();
     }
+
+    virtual future<> on_after_service_level_change(service_level_options slo_before, service_level_options slo_after, service_level_info sl_info) override {
+        return make_ready_future<>();
+    }
 };
 
 std::ostream& operator<<(std::ostream& os, const add_op& op) {
```

Comment:
I would expect the methods will do something to validate that the apis are called at the right moment (for example validating that while running in an "on_before" method the service level is still unchanged / added /removed and in the after the expected change has happened.

Reply:
Honestly I don't quite understand why and how to do it? 
Do you mean to for instance add assert(if service level is in configuration) before this line
https://github.com/scylladb/scylladb/blob/master/service/qos/service_level_controller.cc#L223

Reply:
Added